### PR TITLE
fix: Tests for Latest Version of Mobile App

### PR DESCRIPTION
**Pull request by rupam-seal**
### <samp>🤖 Generated by Copilot at 407fa90</samp>

Fixed some test errors and formatting issues in `personal-cards-matched-expenses.page.spec.ts` and `approver-dialog.component.spec.ts`. Added a mock input property to test the `ApproverDialogComponent` input events.

### <samp>🤖 Generated by Copilot at 407fa90</samp>

> _Mocking `chipInput`_
> _Testing dialog component_
> _Winter of errors_

### <samp>🤖 Generated by Copilot at 407fa90</samp>

*  Fix type and syntax errors in personal-cards-matched-expenses.page.spec.ts ([link](https://github.com/fylein/fyle-mobile-app/pull/2240/files?diff=unified&w=0#diff-6cb32528fa53945eb3d8b8f205e8eaabc40dd7e282dbe251791d1379e78d0c0bL28-R28), [link](https://github.com/fylein/fyle-mobile-app/pull/2240/files?diff=unified&w=0#diff-6cb32528fa53945eb3d8b8f205e8eaabc40dd7e282dbe251791d1379e78d0c0bL110-R110))
*  Add mock value for chipInput property in approver-dialog.component.spec.ts ([link](https://github.com/fylein/fyle-mobile-app/pull/2240/files?diff=unified&w=0#diff-5ab299b6dab74d69840fa8d7ccba3fc88839f755be62402c907b637c1c03333eR122))
*  Format code with trailing commas in both spec files using Prettier ([link](https://github.com/fylein/fyle-mobile-app/pull/2240/files?diff=unified&w=0#diff-6cb32528fa53945eb3d8b8f205e8eaabc40dd7e282dbe251791d1379e78d0c0bL81-R81), [link](https://github.com/fylein/fyle-mobile-app/pull/2240/files?diff=unified&w=0#diff-5ab299b6dab74d69840fa8d7ccba3fc88839f755be62402c907b637c1c03333eL160-R161), [link](https://github.com/fylein/fyle-mobile-app/pull/2240/files?diff=unified&w=0#diff-5ab299b6dab74d69840fa8d7ccba3fc88839f755be62402c907b637c1c03333eL189-R190), [link](https://github.com/fylein/fyle-mobile-app/pull/2240/files?diff=unified&w=0#diff-5ab299b6dab74d69840fa8d7ccba3fc88839f755be62402c907b637c1c03333eL352-R353), [link](https://github.com/fylein/fyle-mobile-app/pull/2240/files?diff=unified&w=0#diff-5ab299b6dab74d69840fa8d7ccba3fc88839f755be62402c907b637c1c03333eL362-R363))

## Clickup
https://app.clickup.com/t/85ztqkbed

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/personal-cards-matched-expenses/personal-cards-matched-expenses.page.spec.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ describe('PersonalCardsMatchedExpensesPage', () => {
   const data: Navigation = {
     extras: { state: { txnDetails: apiPersonalCardTxnsRes.data[0] } },
     id: 0,
-    initialUrl: '',
+    initialUrl: new UrlTree(),
```
Comment (rupam-seal) [inline src/app/fyle/personal-cards-matched-expenses/personal-cards-matched-expenses.page.spec.ts:28]: Screenshot 2023-08-08 at 1 17 41 PM

File: src/app/shared/components/fy-approver/add-approvers-popover/approver-dialog/approver-dialog.component.spec.ts [hunk 2]
```diff
@@ -119,6 +119,7 @@ describe('ApproverDialogComponent', () => {
     const ev: MatChipInputEvent = {
       value: 'label',
       input: getElementBySelector(fixture, '.selection-modal--form-input') as HTMLInputElement,
+      chipInput: null,
```
Comment (rupam-seal) [inline src/app/shared/components/fy-approver/add-approvers-popover/approver-dialog/approver-dialog.component.spec.ts:122]: Screenshot 2023-08-08 at 1 17 41 PM copy

File: src/test.ts [hunk 3]
```diff
@@ -10,7 +10,3 @@ declare const require: any;
 getTestBed().initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting(), {
   teardown: { destroyAfterEach: false },
 });
-// Then we find all the tests.
-const context = require.context('./', true, /\.spec\.ts$/);
-// And load the modules.
-context.keys().map(context);
```
Comment (rupam-seal) [inline src/test.ts:16]: Screenshot 2023-08-08 at 1 25 02 PM

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 4]
```diff
@@ -447,7 +447,7 @@ describe('SplitExpensePage', () => {
       spyOn(component, 'getTotalSplitAmount');
       component.amount = 2000;
       const splitExpenseForm1 = cloneDeep(splitExpenseFormData1);
-      const otherSplitExpenseForm = cloneDeep(splitExpenseFormData6);
+      const otherSplitExpenseForm: FormGroup<any> = cloneDeep(splitExpenseFormData6);
```
Comment (rupam-seal) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:450]: Screenshot 2023-08-16 at 8 41 13 AM

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 5]
```diff
@@ -353,7 +361,7 @@ describe('SplitExpensePage', () => {
         category: new FormControl(''),
       });
 
-      const otherSplitExpenseForm = new FormGroup({
+      const otherSplitExpenseForm: FormGroup = new FormGroup({
```
Comment (rupam-seal) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:364]: Screenshot 2023-08-16 at 8 41 13 AM

File: src/app/fyle/add-edit-advance-request/fy-currency/fy-currency-choose-currency/fy-currency-choose-currency.component.html [hunk 6]
```diff
@@ -11,7 +11,7 @@
     </ion-buttons>
   </ion-toolbar>
   <ion-toolbar class="selection-modal--form-field-container">
-    <mat-form-field class="selection-modal--form-field mat-form-field-no-padding" appearance="none">
```
Comment (rupam-seal) [inline src/app/fyle/add-edit-advance-request/fy-currency/fy-currency-choose-currency/fy-currency-choose-currency.component.html:14]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/shared/components/dependent-fields/dependent-field/dependent-field-modal/dependent-field-modal.component.html [hunk 7]
```diff
@@ -11,7 +11,7 @@
     </ion-buttons>
   </ion-toolbar>
   <ion-toolbar class="text-center">
-    <mat-form-field appearance="none" class="dependent-field-modal__form-field mat-form-field-no-padding">
```
Comment (rupam-seal) [inline src/app/shared/components/dependent-fields/dependent-field/dependent-field-modal/dependent-field-modal.component.html:14]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/shared/components/fy-approver/add-approvers-popover/approver-dialog/approver-dialog.component.html [hunk 8]
```diff
@@ -16,7 +16,6 @@
   <ion-toolbar class="text-center" [ngClass]="{ 'selection-modal--toolbar': selectedApproversList?.length > 0 }">
     <mat-form-field
       class="selection-modal--form-container mat-form-field-no-padding"
-      appearance="none"
```
Comment (rupam-seal) [inline src/app/shared/components/fy-approver/add-approvers-popover/approver-dialog/approver-dialog.component.html:19]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/shared/components/fy-multiselect/fy-multiselect-modal/fy-multiselect-modal.component.html [hunk 9]
```diff
@@ -18,7 +18,6 @@
   <ion-toolbar class="text-center" [ngClass]="{ 'selection-modal--toolbar': currentSelections?.length > 0 }">
     <mat-form-field
       class="selection-modal--form-container mat-form-field-no-padding"
-      appearance="none"
```
Comment (rupam-seal) [inline src/app/shared/components/fy-multiselect/fy-multiselect-modal/fy-multiselect-modal.component.html:21]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/shared/components/fy-select-project/fy-select-modal/fy-select-project-modal.component.html [hunk 10]
```diff
@@ -11,7 +11,7 @@
     </ion-buttons>
   </ion-toolbar>
   <ion-toolbar class="text-center">
-    <mat-form-field class="selection-modal--form-field mat-form-field-no-padding" appearance="none">
```
Comment (rupam-seal) [inline src/app/shared/components/fy-select-project/fy-select-modal/fy-select-project-modal.component.html:14]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/shared/components/fy-select-vendor/fy-select-modal/fy-select-vendor-modal.component.html [hunk 11]
```diff
@@ -11,7 +11,7 @@
     </ion-buttons>
   </ion-toolbar>
   <ion-toolbar class="text-center">
-    <mat-form-field class="selection-modal--form-field mat-form-field-no-padding" appearance="none">
```
Comment (rupam-seal) [inline src/app/shared/components/fy-select-vendor/fy-select-modal/fy-select-vendor-modal.component.html:14]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/shared/components/fy-select/fy-select-modal/fy-select-modal.component.html [hunk 12]
```diff
@@ -13,7 +13,7 @@
     </ion-buttons>
   </ion-toolbar>
   <ion-toolbar *ngIf="enableSearch" class="text-center">
-    <mat-form-field appearance="none" class="selection-modal--form-field mat-form-field-no-padding">
```
Comment (rupam-seal) [inline src/app/shared/components/fy-select/fy-select-modal/fy-select-modal.component.html:16]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/shared/components/fy-userlist/fy-userlist-modal/fy-userlist-modal.component.html [hunk 13]
```diff
@@ -16,7 +16,6 @@
   <ion-toolbar class="text-center" [ngClass]="{ 'selection-modal--toolbar': currentSelections?.length > 0 }">
     <mat-form-field
       class="selection-modal--form-container mat-form-field-no-padding"
-      appearance="none"
```
Comment (rupam-seal) [inline src/app/shared/components/fy-userlist/fy-userlist-modal/fy-userlist-modal.component.html:19]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/shared/components/virtual-select/virtual-select-modal/virtual-select-modal.component.html [hunk 14]
```diff
@@ -13,7 +13,7 @@
     </ion-buttons>
   </ion-toolbar>
   <ion-toolbar *ngIf="enableSearch" class="text-center">
-    <mat-form-field appearance="none" class="selection-modal--form-field mat-form-field-no-padding">
```
Comment (rupam-seal) [inline src/app/shared/components/virtual-select/virtual-select-modal/virtual-select-modal.component.html:16]: Screenshot 2023-08-16 at 11 22 09 AM

File: src/app/fyle/dashboard/tasks/auto-submission-info-card/auto-submission-info-card.component.spec.ts [hunk 15]
```diff
@@ -11,8 +11,9 @@ describe('AutoSubmissionInfoCardComponent', () => {
 
   beforeEach(waitForAsync(() => {
     TestBed.configureTestingModule({
-      declarations: [AutoSubmissionInfoCardComponent, DatePipe],
```
Comment (rupam-seal) [inline src/app/fyle/dashboard/tasks/auto-submission-info-card/auto-submission-info-card.component.spec.ts:14]: Screenshot 2023-08-16 at 11 25 16 AM

File: src/app/fyle/merge-expense/card-transaction-preview/card-transaction-preview.component.spec.ts [hunk 16]
```diff
@@ -15,8 +15,9 @@ describe('CardTransactionPreviewComponent', () => {
 
   beforeEach(waitForAsync(() => {
     TestBed.configureTestingModule({
-      declarations: [CardTransactionPreviewComponent, EllipsisPipe, DatePipe, CurrencyPipe],
```
Comment (rupam-seal) [inline src/app/fyle/merge-expense/card-transaction-preview/card-transaction-preview.component.spec.ts:18]: Screenshot 2023-08-16 at 11 25 16 AM

File: src/app/fyle/my-advances/my-advances-card/my-advances-card.component.spec.ts [hunk 17]
```diff
@@ -19,9 +19,10 @@ describe('MyAdvancesCardComponent', () => {
   beforeEach(waitForAsync(() => {
     const advanceRequestServiceSpy = jasmine.createSpyObj('AdvanceRequestService', ['getInternalStateAndDisplayName']);
     TestBed.configureTestingModule({
-      declarations: [MyAdvancesCardComponent, EllipsisPipe, HumanizeCurrencyPipe, DatePipe],
```
Comment (rupam-seal) [inline src/app/fyle/my-advances/my-advances-card/my-advances-card.component.spec.ts:22]: Screenshot 2023-08-16 at 11 25 16 AM

File: src/app/fyle/my-view-report/my-view-report-etxn-card/my-view-report-etxn-card.component.spec.ts [hunk 18]
```diff
@@ -15,8 +15,9 @@ describe('MyViewReportEtxnCardComponent', () => {
 
   beforeEach(waitForAsync(() => {
     TestBed.configureTestingModule({
-      declarations: [MyViewReportEtxnCardComponent, FyCategoryIconComponent, EllipsisPipe, DatePipe, CurrencyPipe],
```
Comment (rupam-seal) [inline src/app/fyle/my-view-report/my-view-report-etxn-card/my-view-report-etxn-card.component.spec.ts:18]: Screenshot 2023-08-16 at 11 25 16 AM

File: src/app/fyle/my-view-report/my-view-report-etxn-card/my-view-report-etxn-card.component.spec.ts [hunk 19]
```diff
@@ -15,8 +15,9 @@ describe('MyViewReportEtxnCardComponent', () => {
...
+      declarations: [MyViewReportEtxnCardComponent, FyCategoryIconComponent, EllipsisPipe],
       imports: [IonicModule.forRoot(), MatIconModule, MatIconTestingModule],
+      providers: [DatePipe, CurrencyPipe],
```
Comment (rupam-seal) [inline src/app/fyle/my-view-report/my-view-report-etxn-card/my-view-report-etxn-card.component.spec.ts:20]: Screenshot 2023-08-16 at 11 25 16 AM

File: src/app/fyle/personal-cards-matched-expenses/personal-cards-matched-expenses.page.spec.ts [hunk 20]
```diff
@@ -36,7 +36,7 @@ describe('PersonalCardsMatchedExpensesPage', () => {
     const modalControllerSpy = jasmine.createSpyObj('ModalController', ['create']);
     const modalPropertiesSpy = jasmine.createSpyObj('ModalPropertiesService', ['getModalDefaultProperties']);
     TestBed.configureTestingModule({
-      declarations: [PersonalCardsMatchedExpensesPage, CurrencyPipe, DatePipe, CurrencySymbolPipe],
```
Comment (rupam-seal) [inline src/app/fyle/personal-cards-matched-expenses/personal-cards-matched-expenses.page.spec.ts:39]: Screenshot 2023-08-16 at 11 25 16 AM

File: src/app/fyle/view-team-report/view-team-report-etxn-card/view-team-report-etxn-card.component.spec.ts [hunk 21]
```diff
@@ -15,8 +15,9 @@ describe('ViewTeamReportEtxnCardComponent', () => {
 
   beforeEach(waitForAsync(() => {
     TestBed.configureTestingModule({
-      declarations: [ViewTeamReportEtxnCardComponent, FyCategoryIconComponent, EllipsisPipe, DatePipe, CurrencyPipe],
```
Comment (rupam-seal) [inline src/app/fyle/view-team-report/view-team-report-etxn-card/view-team-report-etxn-card.component.spec.ts:18]: Screenshot 2023-08-16 at 11 25 16 AM

Comment (rupam-seal) [reply src/app/fyle/view-team-report/view-team-report-etxn-card/view-team-report-etxn-card.component.spec.ts:18]: Screenshot 2023-08-16 at 11 29 55 AM

File: src/app/fyle/personal-cards-matched-expenses/personal-cards-matched-expenses.page.spec.ts (hunk 20 above)
Comment (rupam-seal) [reply src/app/fyle/personal-cards-matched-expenses/personal-cards-matched-expenses.page.spec.ts:39]: Screenshot 2023-08-16 at 11 29 55 AM

File: src/app/fyle/merge-expense/card-transaction-preview/card-transaction-preview.component.spec.ts (hunk 16 above)
Comment (rupam-seal) [reply src/app/fyle/merge-expense/card-transaction-preview/card-transaction-preview.component.spec.ts:18]: Screenshot 2023-08-16 at 11 29 55 AM

File: src/app/fyle/my-advances/my-advances-card/my-advances-card.component.spec.ts (hunk 17 above)
Comment (rupam-seal) [reply src/app/fyle/my-advances/my-advances-card/my-advances-card.component.spec.ts:22]: Screenshot 2023-08-16 at 11 29 55 AM

File: src/app/shared/components/fy-filter-pills/fy-filter-pills.component.spec.ts [hunk 22]
```diff
@@ -12,7 +12,7 @@ describe('FyFilterPillsComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [FyFilterPillsComponent, SnakeCaseToSpaceCase, TitleCasePipe],
```
Comment (rupam-seal) [inline src/app/shared/components/fy-filter-pills/fy-filter-pills.component.spec.ts:15]: Screenshot 2023-08-16 at 11 41 59 AM

File: src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.spec.ts [hunk 23]
```diff
@@ -15,7 +15,7 @@ describe('SidemenuHeaderComponent', () => {
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      declarations: [SidemenuHeaderComponent, InitialsPipe, UpperCasePipe, TitleCasePipe],
```
Comment (rupam-seal) [inline src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.spec.ts:18]: Screenshot 2023-08-16 at 11 41 59 AM

Comment (rupam-seal) [reply src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.spec.ts:18]: Screenshot 2023-08-16 at 11 43 48 AM

**Review by Dimple16 — APPROVED** [review]
the changes lgtm, but we'll have to test all the modals thoroughly,

**Review by OmkarJ13 — APPROVED** [review]

**Pull request closed without merging**